pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -122,11 +123,28 @@ func (uc *UpgradeCluster) getClusterNodeStatus(az armhelpers.AKSEngineClient, re
 		kubeClient = k
 	}
 
+	clusterID := uc.DataModel.Properties.GetClusterID()
+	agentPoolsToUpgrade := make(map[string]bool)
+	for agentPoolName, val := range uc.AgentPoolsToUpgrade {
+		agentPoolsToUpgrade[agentPoolName] = val
+	}
+
 	for vmScaleSetPage, err := uc.Client.ListVirtualMachineScaleSets(ctx, resourceGroup); vmScaleSetPage.NotDone(); err = vmScaleSetPage.NextWithContext(ctx) {
 		if err != nil {
 			return err
 		}
 		for _, vmScaleSet := range vmScaleSetPage.Values() {
+			vmssRegex := regexp.MustCompile(utils.VMSSNamingFormat)
+			if capture := vmssRegex.FindStringSubmatch(*vmScaleSet.Name); capture != nil {
+				if !strings.EqualFold(clusterID, capture[utils.VMSSClusterIDIndex]) {
```

Comment:
(re-adding my comment) I think this might not work for Windows because windows nodes only have a substring of the Cluster ID in their name

Reply:
sure, missed that..I some how started living in the new world where naming is the same.. will add the check for windows.. and resend...

Reply:
I would really prefer to avoid regex because it makes the pattern cryptic, especially that is easy to just split on a character in this case.